test/basic.html

```diff
@@ -57,110 +57,122 @@
-          g2.addEventListener('paper-radio-group-changed', function () {
-            expect(items[expectedCheckedItem].checked).to.be.equal(true);
-
-            // Right arrow should go to the next item.
-            expectedCheckedItem = 1;
-            var event = new CustomEvent('keydown');
-            event.keyCode = event.code = 39;
-            g2.dispatchEvent(event);
-
-            done();
-          });
+          var g = fixture('WithSelection');
+          var items = g.items;
+
+          setTimeout(function() {
+            // Initially, the first item is checked.
+            expect(items[0].checked).to.be.equal(true);
+
+            g.addEventListener('paper-radio-group-changed', function () {
+              expect(items[1].checked).to.be.equal(true);
+              done();
+            });
+            MockInteractions.keyDownOn(g, RIGHT_ARROW);
+          // Wait for async in ready().
+          }, 200);
         });
 
         test('left arrow reverses the selection', function (done) {
-          var items = g2.items;
-
-          // Initially, the first item is checked.
-          var expectedCheckedItem = 0;
-          g2.addEventListener('paper-radio-group-changed', function () {
-            expect(items[expectedCheckedItem].checked).to.be.equal(true);
+          var g = fixture('WithSelection');
+          var items = g.items;
 
-            // Left arrow should go to the last item.
-            expectedCheckedItem = 2;
-            var event = new CustomEvent('keydown');
-            event.keyCode = event.code = 37;
-            g2.dispatchEvent(event);
+          setTimeout(function() {
+            // Initially, the first item is checked.
+            expect(items[0].checked).to.be.equal(true);
 
-            done();
-          });
+            g.addEventListener('paper-radio-group-changed', function () {
+              expect(items[2].checked).to.be.equal(true);
+              done();
+            });
+            MockInteractions.keyDownOn(g, LEFT_ARROW);
+          // Wait for async in ready().
+          }, 200);
         });
 
         test('selection should skip disabled items', function (done) {
-          var items = g3.items;
-
-          // Initially, the first item is checked.
-          var expectedCheckedItem = 0;
-          g3.addEventListener('paper-radio-group-changed', function () {
-            expect(items[expectedCheckedItem].checked).to.be.equal(true);
-
-            // Right arrow should go to the last item, since the second item is
-            // disabled.
-            expectedCheckedItem = 2;
-            var event = new CustomEvent('keydown');
-            event.keyCode = event.code = 39;
-            g3.dispatchEvent(event);
-
-            done();
-          });
+          var g = fixture('WithDisabled');
+          var items = g.items;
+
+          setTimeout(function() {
+            // Initially, the first item is checked.
+            expect(items[0].checked).to.be.equal(true);
```

Comment:
If these async expectations throw, no error will be reported. In order to properly report what expectation failed, you need to wrap every expectation with a try / catch and call `done` with the error:

```javascript
try {
  expect(true).to.be.equal(false);
  done();
} catch (e) {
  done(e);
}
```

An alternative to doing lots of (potentially nested) try / catch blocks is to use Promises, which allow you to condence multiple asynchronous exceptions down to one "catch":

```javascript
thisIsAPromiseInstance.then(function() {
  expect(true).to.be.equal(false);
  return somePromiseYieldingOperation();
}).then(function() {
  expect(false).to.be.equal(true);
  done();
}).catch(function(error) {
  done(error);
});
```